Orange/widgets/data/tests/test_owdatasampler.py

```diff
@@ -81,6 +81,42 @@ def test_no_intersection_in_outputs(self):
                     self.assertEqual(len(self.zoo), len(sample) + len(other))
                     self.assertNoIntersection(sample, other)
 
+    def test_bigger_size_with_replacement(self):
+        """Allow bigger output without replacement."""
+        _, out_size, actual_size = self._set_bigger_sample_size(True)
+        self.assertEqual(out_size, actual_size,
+                         'Sample size should not be lowered with replacement')
+
+    def test_bigger_size_without_replacement(self):
+        """Lower output samples to match input's without replacement."""
+        in_size, _, actual_size = self._set_bigger_sample_size(False)
+        self.assertEqual(in_size, actual_size)
+
+    def test_bigger_output_warning(self):
+        self._set_bigger_sample_size(with_replacement=True)
+        self.assertTrue(self.widget.Warning.bigger_sample.is_shown())
+
+    def _set_bigger_sample_size(self, with_replacement):
+        """Load data, set sample size and click to generate samples.
+
+        The sample size is set to out_size which is bigger than the input size
+        in_size.
+
+        Returns integers in_size, out_size and the actual gui sample size.
+        """
+        in_size = 2
+        out_size = in_size + 1
+        data = self.iris[:in_size]
+        self.widget.set_data(data)
```

Comment:
We try to avoid calling input handlers directly and use self.send_signal instead.
```
self.send_signal(self.widget.Inputs.data, data)
```